service/storage_service.cc

```diff
@@ -1992,8 +1992,12 @@ class topology_coordinator {
                 // node into `exclude_nodes` parameter in `exec_global_command`, but CDC generations are never
                 // introduced during replace/remove.
                 {
+                    raft_topology_cmd cmd{raft_topology_cmd::command::barrier};
+                    for (const auto& [id, _]: _topo_sm._topology.transition_nodes) {
+                        cmd.wait_for_ip.push_back(id);
+                    }
                     auto f = co_await coroutine::as_future(exec_global_command(std::move(guard),
-                        raft_topology_cmd::command::barrier,
+                        cmd,
```

Comment:
replace does not pass through commit_cdc_generation, but it should be done for replace as well, no?

Reply:
> replace does not pass through commit_cdc_generation

hmm, I hoped for that. Should we put the similar barrier for replace?

Reply:
Replace goes to `topology::transition_state::tablet_draining` and there is no guaranty this state will call a barrier (only if there are tablets to drain). I guess we can add explicit barrier call there.

Reply:
Can't we do this `wait_for_ip` barrier inside `topology::transition_state::join_group0` which should always be the first step in bootstrap/replace?

Reply:
Added `wait_for_ip` command. We need to call it from `handle_node_transition`, before moving topology to transition, otherwise `topology_state_load` could observe ids without ips